go/mysql/flavor.go

```diff
@@ -457,23 +457,23 @@ func parseReplicationStatus(fields map[string]string) ReplicationStatus {
 		SQLState:     ReplicationStatusToState(fields["Slave_SQL_Running"]),
 		LastSQLError: fields["Last_SQL_Error"],
 	}
-	parseInt, _ := strconv.ParseInt(fields["Master_Port"], 10, 0)
-	status.SourcePort = int(parseInt)
-	parseInt, _ = strconv.ParseInt(fields["Connect_Retry"], 10, 0)
-	status.ConnectRetry = int(parseInt)
-	parseUint, err := strconv.ParseUint(fields["Seconds_Behind_Master"], 10, 0)
+	parseInt, _ := strconv.ParseInt(fields["Master_Port"], 10, 32)
+	status.SourcePort = int32(parseInt)
+	parseInt, _ = strconv.ParseInt(fields["Connect_Retry"], 10, 32)
+	status.ConnectRetry = int32(parseInt)
+	parseUint, err := strconv.ParseUint(fields["Seconds_Behind_Master"], 10, 32)
 	if err != nil {
-		// we could not parse the value into a valid uint -- most commonly because the value is NULL from the
+		// we could not parse the value into a valid uint32 -- most commonly because the value is NULL from the
 		// database -- so let's reflect that the underlying value was unknown on our last check
 		status.ReplicationLagUnknown = true
 	} else {
 		status.ReplicationLagUnknown = false
-		status.ReplicationLagSeconds = uint(parseUint)
+		status.ReplicationLagSeconds = uint32(parseUint)
 	}
-	parseUint, _ = strconv.ParseUint(fields["Master_Server_Id"], 10, 0)
-	status.SourceServerID = uint(parseUint)
-	parseUint, _ = strconv.ParseUint(fields["SQL_Delay"], 10, 0)
-	status.SQLDelay = uint(parseUint)
+	parseUint, _ = strconv.ParseUint(fields["Master_Server_Id"], 10, 32)
+	status.SourceServerID = uint32(parseUint)
+	parseUint, _ = strconv.ParseUint(fields["SQL_Delay"], 10, 32)
```

Comment:
nit: I would suggest to create a constant for '32' .. e.g const int32Size = 32

Reply:
@rsajwani That seems overkill to me for stuff like this. It's about this specific value and we have `ParseInt` and related functions in many places.

Reply:
that was the idea.. its used at alot of place.. a minor change in future will result us to change it everywhere .. strconv already have const define like these.. But I am ok if you fee its overkill

Reply:
Integer type sizes will never change :smile:.